File: library/src/main/java/com/yzyfdf/library/rx/BaseRespose.java
package com.yzyfdf.library.rx;

import java.io.Serializable;

public class BaseRespose<T> implements Serializable {
    private int    code;
    private String message;
    private T data;

    public boolean success() {
        return this.code == 0;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    @Override
    public String toString() {
        return "BaseRespose{" +
                "code=" + code +
                ", message='" + message + '\'' +
                ", data=" + data +
                '}';
    }
}

File: library/src/main/java/com/yzyfdf/library/rx/BaseRespose.kt
package com.yzyfdf.library.rx

import java.io.Serializable

class BaseRespose<T> : Serializable {
    var code: Int = 0
    var message: String? = null
    var data: T? = null

    fun success(): Boolean {
        return this.code == 0
    }

    override fun toString(): String {
        return "BaseRespose{" +
                "code=" + code +
                ", message='" + message + '\''.toString() +
                ", data=" + data +
                '}'.toString()
    }
}
